lib/fluent/plugin/filter_suppress.rb

```diff
@@ -27,21 +30,45 @@ def filter_stream(tag, es)
         end
         slot = @slots[key] ||= []
 
-        # expire old records time
-        expired = time.to_f - @interval
+        # expire old records record_time
+        expired = record_time.to_f - @interval
         while slot.first && (slot.first <= expired)
           slot.shift
         end
 
-        if slot.length >= @num
+        if should_suppress?(record_slot: slot, number_to_keep: @num)
           log.debug "suppressed record: #{record.to_json}"
+          suppressed_count += 1
           next
         end
 
-        slot.push(time.to_f)
-        new_es.add(time, record)
+        slot.push(record_time.to_f)
+        new_es.add(record_time, record)
       end
-      return new_es
+
+      optionally_give_feedback(event_stream: new_es, count: suppressed_count)
+      new_es
+    end
+
+    private
+
+    def optionally_give_feedback(event_stream:, count:)
+      if option_give_feedback? && count > 0
+        log.debug "Suppressed #{count} records"
+        event_stream.add(current_time, {'message' => "And #{count} more..."})
```

Comment:
I think feedback message is better to be structural including count,  etc. values because a structural message is friendly with machine processing.
e.g.
```ruby
{
  'message' => "And #{count} more...",
  'plugin_id' => @id,
  'suppressed' => true,
  'count' => count,
}
```

Reply:
Great idea. I'll add that.